File: PanKanapka/Api.Domain/Models/TaskItem.cs
using System.Collections.Generic;

namespace Api.Domain.Models
{
    public class FirmTask
    {
        public string LogoUrl { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public bool IsActiveTask { get; set; }
    }

    public class TaskItem
    {
        public string Date { get; set; }
        public IEnumerable<FirmTask> Firms { get; set; }
    }
}

File: pankan_appka/lib/api/models/task_item.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pankan_appka/api/models/firm_task.dart';

part 'task_item.g.dart';

@JsonSerializable()
class TaskItem {
 final DateTime date;
 final Iterable<FirmTask> firms;
 TaskItem(this.date, this.firms);

  static fromJson(Map<String, dynamic> jsonMap) => _$TaskItemFromJson(jsonMap);

  Map<String, dynamic> toJson() => _$TaskItemToJson(this);
}